frontend_tests/node_tests/popover_menus_data.js

```diff
@@ -0,0 +1,486 @@
+    assert.deepEqual(response, {
+        message_id: 6,
+        stream_id: 1,
+        editability_menu_item: "translated: Edit message",
+        move_message_menu_item: undefined,
+        should_display_mark_as_unread: false,
+        view_source_menu_item: undefined,
+        should_display_collapse: false,
+        should_display_uncollapse: false,
+        should_display_add_reaction_option: false,
+        should_display_edit_history_option: false,
+        should_display_hide_option: false,
+        conversation_time_uri: "http://zulip.zulipdev.com/undefined/near/6",
+        narrowed: false,
+        should_display_delete_option: false,
+        should_display_read_receipts_option: false,
+        should_display_reminder_option: false,
+        should_display_quote_and_reply: false,
+    });
+
+    reset_page_params_user_status();
+    page_params.is_moderator = true;
+    response = popover_menus_data.get_actions_popover_content_context(5);
+    assert.deepEqual(response, {
+        message_id: 5,
+        stream_id: 1,
+        editability_menu_item: "translated: Edit message",
+        move_message_menu_item: "translated: Move messages",
+        should_display_mark_as_unread: true,
+        view_source_menu_item: undefined,
+        should_display_collapse: true,
+        should_display_uncollapse: false,
+        should_display_add_reaction_option: false,
+        should_display_edit_history_option: true,
+        should_display_hide_option: false,
+        conversation_time_uri: "http://zulip.zulipdev.com/undefined/near/5",
+        narrowed: false,
+        should_display_delete_option: true,
+        should_display_read_receipts_option: true,
+        should_display_reminder_option: false,
+        should_display_quote_and_reply: true,
+    });
+
+    reset_page_params_user_status();
+    page_params.is_admin = true;
+    response = popover_menus_data.get_actions_popover_content_context(2);
+    assert.deepEqual(response, {
+        message_id: 2,
+        stream_id: 1,
+        editability_menu_item: "translated: Edit message",
+        move_message_menu_item: "translated: Move messages",
+        should_display_mark_as_unread: true,
+        view_source_menu_item: undefined,
+        should_display_collapse: true,
+        should_display_uncollapse: false,
+        should_display_add_reaction_option: false,
+        should_display_edit_history_option: true,
+        should_display_hide_option: false,
+        conversation_time_uri: "http://zulip.zulipdev.com/undefined/near/2",
+        narrowed: false,
+        should_display_delete_option: true,
+        should_display_read_receipts_option: true,
+        should_display_reminder_option: false,
+        should_display_quote_and_reply: true,
+    });
+});
+
+test("not_my_message_view_actions", () => {
+    set_page_params_no_edit_restrictions();
+    reset_page_params_user_status();
+    // Get message that is only viewable
```

Comment:
I don't understand how this test's setup is different from the next one, for example.